kong/plugins/runscope-log/log.lua

```diff
@@ -0,0 +1,85 @@
+local cjson = require "cjson"
+local url = require "socket.url"
+
+local _M = {}
+
+local HTTPS = "https"
+
+-- Generates http payload .
+-- @param `method` http method to be used to send data
+-- @param `parsed_url` contains the host details
+-- @param `message`  Message to be logged
+-- @return `payload` http payload
+local function generate_post_payload(parsed_url, access_token, message)
+  local body = cjson.encode(message)
+  local payload = string.format(
+    "%s %s HTTP/1.1\r\nHost: %s\r\nConnection: Keep-Alive\r\nAuthorization: Bearer %s\r\nContent-Type: application/json\r\nContent-Length: %s\r\n\r\n%s",
+    "POST", parsed_url.path, parsed_url.host, access_token, string.len(body), body)
+  return payload
+end
+
+-- Parse host url
+-- @param `url`  host url
+-- @return `parsed_url`  a table with host details like domain name, port, path etc
+local function parse_url(host_url)
+  local parsed_url = url.parse(host_url)
+  if not parsed_url.port then
+    if parsed_url.scheme == "http" then
+      parsed_url.port = 80
+     elseif parsed_url.scheme == HTTPS then
+      parsed_url.port = 443
+     end
+  end
+  if not parsed_url.path then
+    parsed_url.path = "/"
+  end
+  return parsed_url
+end
+
+-- Log to a Http end point.
+-- @param `premature`
+-- @param `conf`     Configuration table, holds http endpoint details
+-- @param `message`  Message to be logged
+local function log(premature, conf, message)
+  local ok, err
```

Comment:
Best to check if `premature` is `true`. That is the case when the process has received a signal that terminated it, and if too many timers have to be fired, it will take a while before executing them all and shutting down.

```lua
local function log(...)
if premature then return end
```

This is not yet in `next` but the invalidations branch has this fix for all timers in Kong.